src/main/kotlin/com/terraformation/backend/accelerator/api/ProjectVotingController.kt

```diff
@@ -0,0 +1,114 @@
+import com.terraformation.backend.api.ApiResponse403
+import com.terraformation.backend.api.ApiResponse404
+import com.terraformation.backend.api.SimpleSuccessResponsePayload
+import com.terraformation.backend.api.SuccessResponsePayload
+import com.terraformation.backend.db.accelerator.CohortPhase
+import com.terraformation.backend.db.accelerator.VoteOption
+import com.terraformation.backend.db.default_schema.ProjectId
+import com.terraformation.backend.db.default_schema.UserId
+import io.swagger.v3.oas.annotations.Operation
+import org.springframework.web.bind.annotation.DeleteMapping
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+
+@AcceleratorEndpoint
+@RequestMapping("/api/v1/accelerator/projects/{projectId}/votes")
+@RestController
+class ProjectVotingController() {
+  @ApiResponse200
+  @ApiResponse403
+  @ApiResponse404
+  @GetMapping
+  @Operation(
+      summary = "Gets vote selections for a single project.",
+      description =
+          "List every vote selection for this project, organized by phases. Each phase will " +
+              "contain a list of eligible voters and their selections. If a `voteOption` is " +
+              "`null`, the voter has not yet casted a vote. ")
+  fun getProjectVotes(
+      @PathVariable("projectId") projectId: ProjectId,
+  ): GetProjectVotesResponsePayload {
+    return GetProjectVotesResponsePayload(projectId, emptyList())
+  }
+
+  @ApiResponse200
+  @ApiResponse403
+  @ApiResponse404
+  @PutMapping
+  @Operation(
+      summary = "Upserts vote selections for a single project.",
+      description =
+          "Update the user's vote for the project phase. If the (user, project, phase) does not, " +
+              "exist, a new entry is created. If a `voteOption` is set to `null`, the voter has " +
+              "not yet casted a vote. ")
+  fun upsertProjectVotes(
+      @PathVariable("projectId") projectId: ProjectId,
+      @RequestBody payload: UpsertProjectVotesRequestPayload,
+  ): UpsertProjectVotesResponsePayload {
+    return UpsertProjectVotesResponsePayload(projectId, emptyList())
+  }
+
+  @ApiResponse200
+  @ApiResponse403
+  @ApiResponse404
+  @DeleteMapping
+  @Operation(
+      summary = "Remove one or more voters from the project/phase.",
+      description =
+          "Delete the voters from the project phase, making them ineligible from voting. This is " +
+              "different from undoing a vote (by setting the `voteOption` to `null`). `projectId`" +
+              "must be provided. If neither `phaseId` or `userId` is provided, all voters from " +
+              "every phase are removed. If only `phaseId` is provided, all voters from that " +
```

Comment:
"Default is to delete all voters" makes me a bit nervous.

Do we need bulk voter deletion in the API at all? If so, maybe as a guard against accidents it'd make sense to add a payload field with an enum that you have to explicitly set to say you want to delete a certain set of things, and then validate that the IDs match what the user says they intended.

Reply:
I don't think bulk delete is part of the requirements and it doesn't seem like we will have many use cases for that. I am opting to use a boolean flag for deleting a phase of voters, and not exposing deleting a whole project of voters at all.